api/api.go

```diff
@@ -712,6 +717,18 @@ func NewClient(config *Config) (*Client, error) {
 			return nil, fmt.Errorf("Unknown protocol scheme: %s", parts[0])
 		}
 		config.Address = parts[1]
+
+		// separate out a reverse proxy prefix, if it is present.
+		// NOTE: Rewriting this code to use url.Parse() instead of
+		// strings.SplitN() breaks existing test cases.
+		switch parts[0] {
+		case "http", "https":
+			parts := strings.SplitN(parts[1], "/", 2)
```

Comment:
I spent quite a while trying to refactor and use url.Parse() but when I did, my code worked but I broke various unit tests.

I've come to the conclusion that the authors chose strings.SplitN() in the original code for a reason -- because they wanted to support some inputs that give url.Parse() indigestion.

Reply:
Fair enough. Thanks for adding the note!